test/ui/selenium/LoginTests.js

```diff
@@ -0,0 +1,252 @@
+
+            await homePage.mainPage();
+            await sleep(2000);
+            console.log(await driver.getTitle());
+
+            await homePage.openLoginPage();
+            await sleep(2000);
+            console.log(await driver.getTitle());
+
+            await loginPage.enterEmail(email);
+            await loginPage.enterPassword(password);
+            await loginPage.clickLoginButton();
+            await sleep(3000);
+            console.log(await driver.getTitle());
+
+            let logout_link = await homePage.logoutLink();
+            let getLogout = await logout_link.getText();
+            assert(getLogout == "Log Out");
+
+        } catch (error) {
+            console.log("error during login: ", error);
+            throw error;
+        } 
+    });
+
+    it('@login should fail login with invalid password', async function () {
+        
+        this.timeout(100000);
+
+        try {
+
+            console.log('Test - Invalid Password');
+
+            const homePage = new HomePage(driver);
+            const loginPage = new LoginPage(driver);
+
+
+            await homePage.mainPage();
+            await sleep(2000);
+            console.log(await driver.getTitle());
+
+            await homePage.openLoginPage();
+            await sleep(2000);
+            console.log(await driver.getTitle());
+
+            await loginPage.enterEmail(email);
+            await loginPage.enterPassword('test');
+            await loginPage.clickLoginButton();
+            await sleep(3000);
+            console.log(await driver.getTitle());
+
+            let elm = await driver.findElement(By.xpath("/html/body/div/div/h3"))
+            let expected = await elm.getText();
+
+            assert(expected == "User is not valid: Invalid credentials");
+            
+            await sleep(2500);
+            
+            console.log("Test Passed (Invalid Password), ", expected);
+            
+
+        } catch (error) {
+            console.log("Test Failed");
+            console.log("error during login: ", error);
+            throw error;
+        }
+    })
+
+    it('@login should fail login with invalid email', async function () {
+
+        this.timeout(100000);
+
+        try {
+
+            console.log("Test - Invalid Email")
+
+
+            const homePage = new HomePage(driver);
+            const loginPage = new LoginPage(driver);
+
+
+            await homePage.mainPage();
+            await sleep(2000);
+            console.log(await driver.getTitle());
+
+            await homePage.openLoginPage();
+            await sleep(2000);
+            console.log(await driver.getTitle());
+
+            await loginPage.enterEmail('test@kainos.com');
+            await loginPage.enterPassword(password);
+            await loginPage.clickLoginButton();
+            await sleep(3000);
+            console.log(await driver.getTitle());
+
+            let elm = await driver.findElement(By.xpath("/html/body/div/div/h3"))
```

Comment:
The element should have an id in the html and you should then be able to get it by id. xpath should be avoided when possible

Reply:
will add ID